protoc-gen-grpc-gateway/gengateway/template.go

```diff
@@ -248,6 +248,9 @@ var (
 	handleSend := func() error {
 		var protoReq {{.Method.RequestType.GoType .Method.Service.File.GoPkg.Path}}
 		err = dec.Decode(&protoReq)
+		if err == io.EOF {
```

Comment:
@tmc There's no guarantee that `io.EOF` is/will be coming only from `dec.Decode`.
The best place to deal with an error is the place the error happens.